tests/common/test_sysinfo.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import platform
+
+from azurelinuxagent.common.protocol.imds import ComputeInfo
+from azurelinuxagent.common.protocol.restapi import VMInfo
+from azurelinuxagent.common.sysinfo import SysInfo
+from azurelinuxagent.common.version import AGENT_EXECUTION_MODE, DISTRO_NAME, DISTRO_VERSION, DISTRO_CODE_NAME
+from tests.tools import patch, AgentTestCase, PropertyMock
+
+
+class TestSysInfo(AgentTestCase):
+
+    def setUp(self):
+        AgentTestCase.setUp(self)
+        SysInfo._instance = None
+
+    @staticmethod
+    def build_vminfo():
+        vminfo = VMInfo()
+        vminfo.vmName = "TEST_VMName"
```

Comment:
NIT: You can add all these parameters as part of the constructor itself, no need to do it separately 
```python
class VMInfo(DataContract):
    def __init__(self,
                 subscriptionId=None,
                 vmName=None,
                 containerId=None,
                 roleName=None,
                 roleInstanceName=None,
                 tenantName=None):